### PR TITLE
[CBRD-22954] Recovery of active transactions start stream position

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-22954

End of analysis: find the smallest LSA of all active transactions.
Undo: walk backwards from this LSA until a GC record and keep its stream position

Todo: 
- use this stream position for [CBRD-22955](http://jira.cubrid.org/browse/CBRD-22955)
- fix possible removed archived log case  [CBRD-23016](http://jira.cubrid.org/browse/CBRD-23016)

### CODE REVIEW

File: src/transaction/log_recovery.c [hunk 1]
```diff
@@ -5013,6 +5028,65 @@ log_recovery_undo (THREAD_ENTRY * thread_p)
...
+
+		  // found start of filtered apply
+		  log_Gl.m_active_start_position = gc_rec->stream_pos;
+		  found = true;
+		  break;
+		}
+
+	      if (LSA_ISNULL (&log_rec->back_lsa))
+		{
+		  assert (false);
```
[the remark this answers is not in the document]
Comment (razvanradulescu) [reply src/transaction/log_recovery.c]: Any database has at least createdb log records

File: src/transaction/log_recovery.c [hunk 2]
```diff
@@ -5013,6 +5028,65 @@ log_recovery_undo (THREAD_ENTRY * thread_p)
...
+  /* Find log_Gl m_active_start position located in the last GROUP COMPLETE record before m_min_active_lsa (min pos */
+  /* of last active transaction at the end of analysis) - we go backwards until we find a GC record */
+  if (!LSA_ISNULL (&log_Gl.m_min_active_lsa))
+    {
+      assert (log_Gl.m_active_start_position == 0);
+      bool found = false;
+      LSA_COPY (&log_lsa, &log_Gl.m_min_active_lsa);
+      while (!found)
+	{
+	  if (logpb_fetch_page (thread_p, &log_lsa, LOG_CS_FORCE_USE, log_pgptr) != NO_ERROR)
```
Comment (alexstanro) [inline src/transaction/log_recovery.c]: What if the GROUP_COMPLETE is in the log archive and the archive was removed?

File: src/transaction/log_recovery.c [hunk 3]
```diff
@@ -2882,6 +2882,22 @@ log_recovery_analysis (THREAD_ENTRY * thread_p, LOG_LSA * start_lsa, LOG_LSA * s
 	}
     }
 
+  /* Find min pos of last active transaction at the end of analysis */
```
[the remark this answers is not in the document]
Comment (radudoros) [reply src/transaction/log_recovery.c]: logtb_find_smallest_lsa does not seem to fit my purposes, I'd need to add additional check for undo_nxlsa. I would not like to add new flags to logtb_find_smallest_lsa only to addapt for my usage.

File: src/transaction/log_recovery.c (hunk 2 above)
Comment (radudoros) [reply src/transaction/log_recovery.c]: added todo & opened an issue

**Pull request merged**